lib/PuppeteerSharp.Tests/FrameTests/WaitForNavigationTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace PuppeteerSharp.Tests.FrameTests
+{
+    [Collection("PuppeteerLoaderFixture collection")]
+    public class WaitForNavigationTests : PuppeteerPageBaseTest
+    {
+        public WaitForNavigationTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task ShouldWaitForNavigate()
+        {
+            await Page.GoToAsync(TestConstants.ServerUrl + "/frames/one-frame.html");
+            var frame = Page.Frames[1];
+            var waitForNavigationResult = frame.WaitForNavigationAsync();
+
+            await Task.WhenAll(
+                waitForNavigationResult,
+                frame.EvaluateFunctionAsync("url => window.location.href = url", TestConstants.ServerUrl + "/grid.html")
+            );
+            var response = await waitForNavigationResult;
+            Assert.Equal(HttpStatusCode.OK, response.Status);
+            Assert.Contains("grid.html", response.Url);
+            Assert.Equal(frame, response.Frame);

```

Comment:
Let's use same instead of equal
```suggestion
            Assert.Same(frame, response.Frame);
```